contrib/dokku_client.sh

```diff
@@ -1,32 +1,111 @@
+
+function random_number() {
+  [[ -n "$1" ]] && RANGE="$1"
+  if [[ -n "$RANGE" ]];then
+    number=$RANDOM
+    let "number %= $RANGE"
+  else
+    number=$RANDOM
+  fi
+  echo $number
+}
+
+function random_name() {
+ NUM1=$(random_number ${#MOVES[@]})
+ NUM2=$(random_number ${#MOVES[@]})
+ NUM3=$(random_number ${#NAMES[@]})
+
+ UPPER_APPNAME="${MOVES[${NUM1}]}-${MOVES[${NUM2}]}-${NAMES[${NUM3}]}"
+
+ [[ "$BASH_VERSION" =~ 4.* ]] && lower_appname=${UPPER_APPNAME,,}
+ [[ -z "$lower_appname" ]] && lower_appname=$(echo $UPPER_APPNAME | tr '[:upper:]' '[:lower:]')
+ echo "$lower_appname"
+}
+
+function client_help_msg() {
+  echo "==> Configure the DOKKU_HOST environment variable or run $0 from a repository with a git remote named dokku"
+  echo "--> i.e. git remote add dokku dokku@<dokku-host>:<app-name>"
+  exit 20 # exit with specific status. only used in units tests for now
+}
+
+if [[ -z $DOKKU_HOST ]]; then
+  if [[ -d .git ]] || git rev-parse --git-dir > /dev/null 2>&1; then
+    DOKKU_HOST=$(git remote -v 2>/dev/null | grep -Ei "^dokku" | head -n 1 | cut -f1 -d' ' | cut -f2 -d '@' | cut -f1 -d':' 2>/dev/null || true)
+  else
+    client_help_msg
+  fi
+fi
+
+if [[ ! -z $DOKKU_HOST ]]; then
+  function _dokku {
+    appname=""
+    if [[ -d .git ]] || git rev-parse --git-dir > /dev/null 2>&1; then
+      set +e
+      appname=$(git remote -v 2>/dev/null | grep -Ei "^dokku" | head -n 1 | cut -f1 -d' ' | cut -f2 -d':' 2>/dev/null)
+      set -e
+    else
+      echo "This is not a git repository"
+    fi
+
+    if [[ "$appname" != "" ]] && [[ -n "$*" ]]; then
+      case "$1" in
+        apps*|backup*|help|plugins*|ps:restartall|trace|version)
+          true
+          ;;
+        delete)
+          verb="apps:destroy"
```

Comment:
That makes sense but then we should drop create in favor of apps:create too. That probably makes more sense any since no args execution yields dokku help. I'll do that when I get back home.